deprecated_store/index.ts

```
@@ -0,0 +1,396 @@
+const log = (...args: any[]) => {
+  if (SHOW_LOGS) {
+    console.log(...args);
+  }
+};
+
+type Query<T extends any = string, HasSet extends boolean = true> = {
+  data: T | null;
+  isStale: boolean;
+  get: () => Promise<T | null>;
+  invalidate: () => void;
+} & (HasSet extends true ? { set: (newData: T) => Promise<void> } : {});
```

Biome (1.9.4)

[error] Don't use '{}' as a type.

Prefer explicitly define the object shape. '{}' means "any non-nullable value".

(lint/complexity/noBannedTypes)